app/views/contexts/summary/list_collector_block.py

```diff
@@ -0,0 +1,263 @@
+                    response_metadata=self._response_metadata,
+                    schema=self._schema,
+                    location=Location(
+                        list_name=current_list.name,
+                        list_item_id=list_id,
+                        section_id=self._section["id"],
+                    ),
+                    return_to="section-summary",
+                    return_to_block_id=None,
+                ).serialize()
+                for block in blocks
+            ]
+
+            related_answers_dict[list_id] = serialized_blocks
+
+        return related_answers_dict
+
+    def _get_item_label(self, list_name: str) -> Optional[str]:
+        for item in self._section["summary"].get("items"):
+            if item["for_list"] == list_name and item["item_label"]:
+
+                return str(item["item_label"])
+
+    def _get_item_anchor_answer_id(self, list_name: str) -> Optional[str]:
+        for item in self._section["summary"].get("items"):
+            if item["for_list"] == list_name and item["item_anchor_answer_id"]:
+                return f"#{str(item['item_anchor_answer_id'])}"
+
+    def get_blocks_for_related_answers(
+        self, related_answers: tuple[ImmutableDict]
+    ) -> list[Optional[ImmutableDict]]:
+        blocks = []
+        answers_by_block = defaultdict(list)
+
+        for answer in related_answers:
+            answer_id = answer["identifier"]
+            block = self._schema.get_block_for_answer_id(answer_id)
+
+            block_to_keep = (
+                block["edit_block"]  # type: ignore
+                # block is not optional at this point
+                if block["type"] == "ListCollector"  # type: ignore
+                else block
+            )
+            answers_by_block[block_to_keep].append(answer_id)
+
+        for immutable_block, answer_ids in answers_by_block.items():
+            block = self._schema.get_mutable_deepcopy(immutable_block)
+
+            # We need to filter out answers for both variants and normal questions
+            for question in block.get("question_variants", [block.get("question")]):  # type: ignore
+                # block is not optional at this point
+                if question.get("answers"):
+                    answers = [
+                        answer
+                        for answer in question["answers"]
+                        if answer["id"] in answer_ids
+                    ]
+                    # Mutate the answers to only keep the related answers
+                    question["answers"] = answers
+                else:
+                    answers = [
+                        answer
+                        for answer in question["question"].get("answers")
+                        if answer["id"] in answer_ids
+                    ]
+                    # Mutate the answers to only keep the related answers
+                    question["question"]["answers"] = answers
```

Comment:
It's for variants, our code we ended up with was not digging the answers out as questions in variants is nested differently so I had to make this change for the build and tests to pass. Otherwise summary was breaking but we didn't know that without schema. I only found out after adding one.

Reply:
In that case it can be simplief by updated the `for loop` to:
```py
            # We need to filter out answers for both variants and normal questions
            for variant_or_block in block.get("question_variants", [block]):  # type: ignore
                # block is not optional at this point
                answers = [
                    answer
                    for answer in variant_or_block["question"].get("answers", {})
                    if answer["id"] in answer_ids
                ]
                # Mutate the answers to only keep the related answers
                variant_or_block["question"]["answers"] = answers

            blocks.append(block)
```

Reply:
Thanks I used it except I changed the name to `mutable_block` otherwise mypy is confusing it with previously defined `block`.